File: beginner_tutorials/src/talker.cpp
#include "ros/ros.h"
#include "beginner_tutorials/Song.h"
#include <iostream>
#include <string>

int main(int argc, char *argv[])
{

	ros::init(argc, argv, "talker");
	ros::NodeHandle nh;
	ros::Publisher pub = nh.advertise<beginner_tutorials::Song>("chatter", 10);

	std::string lyrics[3] = {"A","AB","ABC"};
	int count = 0;

	beginner_tutorials::Song msg;
	msg.name = "HJ";
    msg.singger = "JL";

	ros::Rate loop_rate(1);
	while (ros::ok())
	{
		msg.lyrics = lyrics[count % 3];
        count += 1;
        pub.publish(msg);
		ros::spinOnce();
		loop_rate.sleep();
	}

	return 0;
}

File: beginner_tutorials/scripts/talker.py
#!/usr/bin/env python      
# -*- coding: utf-8 -*-

import rospy
from beginner_tutorials.msg import Song

def talker():
    rospy.init_node('talker', anonymous = False )
    pub = rospy.Publisher("chatter", Song, queue_size=10)

    lyrics = ['A','AB','ABC']
    count = 0

    msg = Song()
    msg.name = 'HJ'
    msg.singger = 'JL'

    loop_rate = rospy.Rate(1)
    while not rospy.is_shutdown():
        msg.lyrics = lyrics[count % 3]
        count += 1
        pub.publish(msg)
        loop_rate.sleep()


if __name__ == '__main__':
    try:
        talker()
    except rospy.ROSInterruptException:
        pass